### ISSUE TITLE
BUG(select): Dropdowns are not accessible on iOS devices with VoiceOver. Critical 

### ISSUE THREAD

**Issue by Joeh89** [angular/components#22509]
#### Reproduction
URL: https://material.angular.io/components/select/overview
Version: 11.2.9
All mat-select dropdowns. 

Steps to reproduce:
1. Open VoiceOver on an iOS device and swipe to the first mat-select dropdown. 
2. Double-tapping to expand, and swipe right to go to the first option (or if an option is already selected, focus usually starts there).
 

#### Expected Behavior
1. On step 1: VO should read: Dropdown label + combobox (+ selected value)
2. On step 2: Dropdown should expand and VO should announce that the dropdown is expanded + focus should go directly on the first option (or selected option if a value is already selected).

#### Actual Behavior

1. a) On step 1: VO only reads Dropdown label.
    b) If float-label is set to never, then VO skips entirely the dropdown, non-sighted users would never know that the dropdown exists.
2. Once dropdown is expanded, VO doesn't announce it, when user swipes right, focus moves in the background instead of the options inside the expanded listbox.

#### Environment

- Angular: Mat angular 11.2.9
- CDK/Material: Mat angular 11.2.9
- Browser(s): Safari
- Operating System (e.g. Windows, macOS, Ubuntu): iOS

**Comment by jelbourn** [comment]
Duplicate of #11106

**Comment by Joeh89** [comment]
Thanks @jelbourn , there are 2 other aspects that are not part of the  11106 issue:
1- regarding the float-label, if it's set to never, the dropdown is skipped entirely. (iOS)
2- the role isn't read, somehow when we focus on the dropdown with VO on (iOS), it focuses uniquely on the label, so VO does not inform the user that this is a Dropdown, the user would just think that it's purely content.

**Comment by jelbourn** [comment]
@Joeh89 could you open separate issues for these?

**Comment by Joeh89** [comment]
@jelbourn , just raised the new issue: #22524
